### PR TITLE
Cv 225 employer account authorization

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Authentication/IUserService.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace SFA.DAS.EmployerCommitmentsV2.Web.Authentication
+{
+    public interface IUserService
```
Comment (elliotritchie) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Authentication/IUserService.cs]: What do you think to renaming this to `IAuthenticationService` to be consistent with `SFA.DAS.ProviderCommitments`?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/TestAuthController.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        [Route("account/{AccountHashedId}")]
+        [DasAuthorize(EmployerUserRole.Owner)]
+        public IActionResult RoleTest(TestRequest request)
+        {
+            return View("Index");
+        }
+
+        [Route("noaccount")]
```
Comment (elliotritchie) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/TestAuthController.cs]: I think this action might need a `DasAuthorize` attribute?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/ClaimsPrincipalExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Security.Claims;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Extensions
+{
+    public static class ClaimsPrincipalExtensions
+    {
+        public static Guid? UserRef(this ClaimsPrincipal claimsPrincipal)
```
Comment (elliotritchie) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/ClaimsPrincipalExtensions.cs]: Is this method used?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/IIDentityExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Linq;
+using System.Security.Claims;
+using System.Security.Principal;
+using SFA.DAS.EmployerCommitmentsV2.Web.Authentication;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Extensions
+{
+    public static class IIdentityExtensions
+    {
+        public static string Id(this IIdentity identity)
```
Comment (elliotritchie) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/IIDentityExtensions.cs]: Is this method used?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/DependencyResolution/IoC.cs [hunk 5]
```diff
@@ -8,8 +10,11 @@ public static class IoC
     {
         public static void Initialize(Registry registry)
         {
+            registry.IncludeRegistry<AutoConfigurationRegistry>();
```
Comment (elliotritchie) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/DependencyResolution/IoC.cs]: Does `AutoConfigurationRegistry` need to be included here?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Startup/ConfigurationStartup.cs [hunk 6]
```diff
@@ -9,7 +9,10 @@ public static class ConfigurationStartup
         public static IWebHostBuilder ConfigureDasAppConfiguration(this IWebHostBuilder hostBuilder)
         {
             return hostBuilder.ConfigureAppConfiguration(c => c
-                .AddAzureTableStorage(ConfigurationKeys.EmployerCommitmentsV2));
+                .AddAzureTableStorage(ConfigurationKeys.EmployerCommitmentsV2)
+                .AddAzureTableStorage(ConfigurationKeys.Encoding)
+                .AddAzureTableStorage(ConfigurationKeys.EmployerAccountsApiClient)
```
Comment (elliotritchie) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Startup/ConfigurationStartup.cs]: Are the `EmployerAccountsApiClient` & `EmployerAccountsReadStore` configs needed?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/TestAuthController.cs (hunk 2 above)
Comment (Pauljgraham) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/TestAuthController.cs]: No it's a test to make sue DasAuthorize is not called

File: src/SFA.DAS.EmployerCommitmentsV2.Web/DependencyResolution/IoC.cs (hunk 5 above)
Comment (Pauljgraham) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/DependencyResolution/IoC.cs]: removed

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Authentication/IUserService.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Authentication/IUserService.cs]: I actually think calling it a 'UserService' is more correct as it's not an authentication service, as the user is not necessarily authenticated when it gets called

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/IIDentityExtensions.cs (hunk 4 above)
Comment (Pauljgraham) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/IIDentityExtensions.cs]: removed

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Startup/ConfigurationStartup.cs (hunk 6 above)
Comment (Pauljgraham) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Startup/ConfigurationStartup.cs]: removed

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Authentication/IUserService.cs (hunk 1 above)
Comment (elliotritchie) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Authentication/IUserService.cs]: Yeah it's a tough one to name. It's called `IAuthenticationService` in a few projects now though so I think consistency should win here.

**Pull request merged**